internal/services/mobilenetwork/mobile_network_data_network_resource.go

```diff
@@ -0,0 +1,240 @@
+	Location                     string            `tfschema:"location"`
+	Tags                         map[string]string `tfschema:"tags"`
+}
+
+type DataNetworkResource struct{}
+
+var _ sdk.ResourceWithUpdate = DataNetworkResource{}
+
+func (r DataNetworkResource) ResourceType() string {
+	return "azurerm_mobile_network_data_network"
+}
+
+func (r DataNetworkResource) ModelObject() interface{} {
+	return &DataNetworkModel{}
+}
+
+func (r DataNetworkResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return datanetwork.ValidateDataNetworkID
+}
+
+func (r DataNetworkResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"mobile_network_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: mobilenetwork.ValidateMobileNetworkID,
+		},
+
+		"description": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"location": commonschema.Location(),
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (r DataNetworkResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r DataNetworkResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 180 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model DataNetworkModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			client := metadata.Client.MobileNetwork.DataNetworkClient
+			mobileNetworkId, err := mobilenetwork.ParseMobileNetworkID(model.MobileNetworkMobileNetworkId)
+			if err != nil {
+				return err
+			}
+
+			id := datanetwork.NewDataNetworkID(mobileNetworkId.SubscriptionId, mobileNetworkId.ResourceGroupName, mobileNetworkId.MobileNetworkName, model.Name)
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for existing %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			properties := datanetwork.DataNetwork{
+				Location:   location.Normalize(model.Location),
+				Properties: &datanetwork.DataNetworkPropertiesFormat{},
+				Tags:       &model.Tags,
+			}
+
+			if model.Description != "" {
+				properties.Properties.Description = &model.Description
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, id, properties); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r DataNetworkResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 180 * time.Minute,
```

Comment:
sanity-checking: does it really take 3 hours to update the description/tags?

Reply:
Actually it wont, while it also calls `PUT` function, I want to keep it as same as create...

Reply:
If there is no need for this to be so long we should reduce it to something reasonable. theres no reason create and update should be the same?

Reply:
okay, reduced it to 30 mins.